scipy/stats/stats.py

```diff
@@ -1088,14 +1091,36 @@ def variation(a, axis=0):
         Axis along which to calculate the coefficient of variation. Default
         is 0. If None, compute over the whole array `a`.
 
+    Returns
+    -------
+    variation : float or array
+        Coefficient ot variation as defined by `s / m` where `s` is the
+        standard deviation and `m` is the mean.
+
+    Notes
+    -----
+    Calculating the coefficient of variation only makes sense for non-symmetric
+    distributions.
+
     References
     ----------
     .. [1] Zwillinger, D. and Kokoska, S. (2000). CRC Standard
        Probability and Statistics Tables and Formulae. Chapman & Hall: New
        York. 2000.
 
+    Examples
+    --------
+    >>> from scipy import stats
+    >>> a = np.linspace(stats.erlang.ppf(0.01, 1), 100)
+    >>> stats.variation(a)
+    0.58889670528253069
+
     """
     a, axis = _chk_asarray(a, axis)
+
+    if a.size == 0:
+        return np.nan
```

Comment:
Just thought about it.
For functions that have an axis keyword, there will be possibly a multidimensional answer. If/When we add `missing`, then there could be some elements that are missing/nan similar to the current `mstats`.

In the same way, the user could give us an array that is empty, `a.size == 0` but has more than one dimension, rows or columns. I vaguely remember similar discussions about the empty behavior on the numpy mailing list.

e.g. variation(np.array([]).reshape(0, 0, 0)

```
>>> np.mean(np.array([]).reshape(0,0,0), axis=0)
array([], shape=(0, 0), dtype=float64)
C:\programs\WinPython-64bit-3.4.3.1\python-3.4.3.amd64\lib\site-packages\numpy\core\_methods.py:59: RuntimeWarning: Mean of empty slice.
  warnings.warn("Mean of empty slice.", RuntimeWarning)
```